gprofiler/java.py

```diff
@@ -358,6 +358,8 @@ def snapshot(self) -> ProcessToStackSampleCounters:
                         results[futures[future]] = result
                 except StopEventSetException:
                     raise
+                except psutil.NoSuchProcess:
```

Comment:
Maybe in this PR it's time to merge all those 3 `snapshot` impls?

Reply:
Removing code duplication is always welcome

Reply:
Yes please share, too much duplication

Reply:
yeah... done. we shoulda done it a while ago